### DISCUSSION TITLE
How to reset/clear the BFormFile input?

### DISCUSSION THREAD

**Discussion by sceee** [cdmoro/bootstrap-vue-3#1222] [Q&A]
How can the `BformFile` input be reset / cleared so "no files are selected"?

I know that the file selection of a HTML input [cannot be written by JS](https://bootstrap-vue-next.github.io/bootstrap-vue-next/docs/components/form-file.html#modifying-the-file-selection) but generally, JS should still be able to set the input `value` to `null` to clear/reset it.

I tried to assign a ref to the input but this does not seem to work:
```typescript
<script setup lang="ts">
const inputRef = ref<HTMLInputElement | null>(null)
const files = ref<File[] | null>(null)

const clearFiles = () => {
    if (inputRef.value) {
        // This does not seem to work
        inputRef.value.value = null
    }
}
</script>

<template>
...
<b-form-file
   ref="inputRef"
   multiple
   v-model="files"
   >
</b-form-file>
<b-button variant="danger" size="sm" class="mt-2 mb-2"  @click="clearFiles">Clear files</b-button>
...
</template>
```

**Comment by sceee** [comment]
I created #1231 to add a `reset()` for this as this was [previously possible with `boostrap-vue`](https://bootstrap-vue.org/docs/components/form-file#clearing-the-file-selection).

**Comment by xvaara** [comment]
The internal input element is now exposed as:
```js
inputRef.value.element
```
